Cpp/odin-views/password.cpp

```diff
@@ -185,5 +186,45 @@ namespace {
         }
     } c_reset_password;
 
+    const class password_hash : public fostlib::urlhandler::view {
+      public:
+        password_hash() : view("odin.password.hash") {}
+
+        std::pair<boost::shared_ptr<fostlib::mime>, int> operator()(
+                const fostlib::json &config,
+                const fostlib::string &path,
+                fostlib::http::server::request &req,
+                const fostlib::host &host) const {
+            if (!config.has_key("hash") || !config.has_key("verify")
+                || !config.has_key("then")) {
+                throw fostlib::exceptions::not_implemented(
+                        __PRETTY_FUNCTION__,
+                        "Must supply both hash, verify and then in the "
+                        "configuration");
+            }
+
+            auto body_str = fostlib::coerce<fostlib::string>(
+                    fostlib::coerce<fostlib::utf8_string>(req.data()->data()));
+            fostlib::json body = fostlib::json::parse(body_str);
+            auto hash_value = fostgres::datum(
+                    config["hash"], std::vector<fostlib::string>{}, body, req);
+            auto verify = fostgres::datum(
+                    config["verify"], std::vector<fostlib::string>{}, body,
+                    req);
+            if (not hash_value || not verify || hash_value != verify) {
+                return respond("Hashing failed", 422);
+            }
+            auto const hash_result = odin::hash_password(
```

Comment:
How to test this line? @KayEss

Reply:
Can you can pass headers into the test through the request constructor. You can get response headers from the response object. Either of those help? Look at the [`request` and `response` objects](https://github.com/KayEss/fost-internet/blob/master/Cpp/include/fost/http.server.hpp)

Reply:
Yeah

Reply:
No, I think those are not help.